diff --git a/src/linear_merging.jl b/src/linear_merging.jl
index bf8e100..8f9890b 100644
--- a/src/linear_merging.jl
+++ b/src/linear_merging.jl
@@ -116,15 +116,15 @@ function compute_errors_fast(X::Array{Float64,2}, y::Array{Float64,1})
       res[ii,jj] = normysq - dot(vec(theta' * A), theta)
     end
     if ii + d - 1 < n
-      Ainv = inv(A)
+      #Ainv = inv(A)
       y2 = X[ii:ii+d-1,:]' * y[ii:ii+d-1]
       for jj = ii+d:n
         currow = vec(X[jj,:])
         normysq += y[jj] *  y[jj]
         A += X[jj,:] * X[jj,:]'
-        Ainv -= (Ainv * currow * currow' * Ainv) / (1 + dot(vec(currow' * Ainv), currow))
+        #Ainv -= (Ainv * currow * currow' * Ainv) / (1 + dot(vec(currow' * Ainv), currow))
         y2 += currow * y[jj]
-        theta = Ainv * y2
+        theta = A \ y2
         res[ii,jj] = normysq - dot(vec(theta' * A), theta)
       end
     end
